Hi Gonzalo, thanks a lot for your reply!  

> Can you explain what` -z multdefs` does exactly for the project today? IIUC this is meant to avoid errors that arise from having the same symbol defined multiple times. This should not be the case today as we link either termbox or ncurses, but not both.

When I try to build the project without this linker flag, I get the following output:
```
cc -Wall -Wextra -Werror -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-qual -Werror -std=c17 -pedantic -DINCL_TERMBOX -O3 -flto -ffast-math -c -o obj/command.o src/command.c
cc -O3 -flto -ffast-math obj/command.o obj/debug.o obj/options.o obj/parser.o obj/plot.o obj/termplot.o obj/ui.o obj/ui_termbox.o -o termplot -lm  -ltermbox
/usr/lib64/gcc/x86_64-suse-linux/15/../../../../x86_64-suse-linux/bin/ld: obj/debug.o (symbol from plugin): in function `start_debug':
(.text+0x0): multiple definition of `debug_enabled'; obj/command.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib64/gcc/x86_64-suse-linux/15/../../../../x86_64-suse-linux/bin/ld: obj/options.o (symbol from plugin): in function `init_structs':
(.text+0x0): multiple definition of `options'; obj/command.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib64/gcc/x86_64-suse-linux/15/../../../../x86_64-suse-linux/bin/ld: obj/parser.o (symbol from plugin): in function `_neg':
(.text+0x0): multiple definition of `debug_enabled'; obj/command.o (symbol from plugin):(.text+0x0): first defined here
[...]
/usr/lib64/gcc/x86_64-suse-linux/15/../../../../x86_64-suse-linux/bin/ld: obj/ui.o (symbol from plugin): in function `update_cmd':
(.text+0x0): multiple definition of `functions'; obj/command.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib64/gcc/x86_64-suse-linux/15/../../../../x86_64-suse-linux/bin/ld: obj/ui.o (symbol from plugin): in function `update_cmd':
(.text+0x0): multiple definition of `debug_enabled'; obj/command.o (symbol from plugin):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:61: termplot] Error 1
```
with more of the same at the [...], which AFAIU suggests that many symbols of the `command.o` object file have multiple definitions. My LSP tells me that the variables defined in the header files (such as `command` in `command.h`) can lead to one-definition-rule violations, so I think that's probably the reason for these errors.
I agree that this is weird and seems unneeded, as AFAIK there aren't any actual re-definitions, but I could not get it to build otherwise...

Hope this somewhat clears things up :)

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ WARNINGS = -Wall -Wextra -Werror -Wshadow -Wstrict-prototypes -Wpointer-arith \
 					 -Wcast-qual
 OPT_FLAGS = -O3 -flto -ffast-math
 CFLAGS = $(WARNINGS) -Werror -std=c23 -pedantic
-LFLAGS = -lm -z muldefs
+LFLAGS = -lm
 TEST_DIR = test/
 SRC_DIR = src/
 UI_SRC_DIR = $(SRC_DIR)ui/

--- a/src/command.c
+++ b/src/command.c
@@ -1,7 +1,10 @@
 #include "command.h"
 
-#include <stdlib.h>
 #include <string.h>
+
+command_history_t command_history;
+char command[CMD_SIZE] = {0};
+unsigned int cursor = 0;
 
 void nothing(__attribute__((unused)) char* arg) {
 }

--- a/src/command.h
+++ b/src/command.h
@@ -10,15 +10,17 @@
 #define CMD_HIST 10
 #define CMD_SIZE 500
 
-struct {
+typedef struct command_history_t {
   // Kind of bounded stack.
   char* commands[CMD_HIST];
   unsigned int last;
   unsigned int first;
-} command_history;
+} command_history_t;
 
-char command[CMD_SIZE];
-unsigned int cursor;
+extern command_history_t command_history;
+
+extern char command[CMD_SIZE];
+extern unsigned int cursor;
 
 void run_command(void);
 void input_command(void);

--- a/src/debug.c
+++ b/src/debug.c
@@ -1,6 +1,7 @@
 #include "debug.h"
 
 FILE * dbg;
+int debug_enabled = 0;
 
 void start_debug(void) {
   if (debug_enabled) {

--- a/src/debug.h
+++ b/src/debug.h
@@ -4,7 +4,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-int debug_enabled;
+extern int debug_enabled;
 
 void start_debug(void);
 void d_print(const char * format, ...);

--- a/src/options.c
+++ b/src/options.c
@@ -1,5 +1,7 @@
 #include "options.h"
 
+struct options_t options;
+
 void init_structs(void) {
   options.x_center = 0.0;
   options.y_center = 0.0;

--- a/src/options.h
+++ b/src/options.h
@@ -4,5 +4,5 @@
 #include "structs.h"
 
 
-struct options_t options;
+extern struct options_t options;
 void init_structs(void);

--- a/src/plot.c
+++ b/src/plot.c
@@ -3,6 +3,8 @@
 #define MAX(x,y) ((x)>(y)?(x):(y))
 #define MIN(x,y) ((x)<(y)?(x):(y))
 
+function functions[MAX_FUNCTIONS];
+
 void fill_plot(struct function* new) {
   new->valid = 0;
   if (new->f.size > 0) {

--- a/src/plot.h
+++ b/src/plot.h
@@ -13,12 +13,14 @@
 
 enum { EXPLICIT, IMPLICIT };
 
-struct function {
+typedef struct function {
     expr f;
     expr g;
     unsigned char type;
     unsigned char valid;
-} functions[MAX_FUNCTIONS];
+} function;
+
+extern function functions[MAX_FUNCTIONS];
 
 
 double test_interval(const expr* e1, const expr* e2, 

--- a/src/ui/ui_ncurses.c
+++ b/src/ui/ui_ncurses.c
@@ -2,6 +2,9 @@
 
 #include <stdlib.h>
 
+WINDOW* win = NULL;
+bool curses_started = false;
+
 void start_ui(void) {
   if (curses_started) {
     refresh();

--- a/src/ui/ui_ncurses.h
+++ b/src/ui/ui_ncurses.h
@@ -9,8 +9,8 @@
 #define FG_WHITE BW
 #define BG_BLACK BW
 
-WINDOW* win;
-bool curses_started;
+extern WINDOW* win;
+extern bool curses_started;
 
 void start_ui(void);
 void end_ui(void);

